### PR TITLE
Stats: Update highlight cards' number format and info architecture

### PR DESCRIPTION
**Pull request by jsnmoon**
## Proposed Changes

- Removes absolute difference but leaves the percentile difference
- Simplify the tooltip content to show "`beforeValue` → `afterValue`".
- Removes unnecessary vertical space below the card icon.
- Enables the cards to shrink/grow depending on content dimensions.

-|Before|After
-|-|-
LG|image|image
SM|image|image
-|image|image
-|image|image
-|image|image

## Why are these changes being made?

The current state of highlight cards is rudimentary and lacks usability polish. This PR increases information density and functionality.

## Testing Instructions

* Open the live branch for this change and ensure that the highlight cards render and behave as expected on the following pages for **desktop and mobile viewports**:
    * Traffic dashboard (`/stats/day/example.com`).
    * Insights dashboard (`/stats/insights/example.com`).
    * Subscribers dashboard (`/stats/subscribers/example.com`).

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by ederrengifo — CHANGES_REQUESTED** [review]
This looks good but I'm concerned about what we show in the tooltip. If the highlight shows the percentage increase/decrease, in the tooltip I would expect to see that difference in numbers, so I think it would be better if we show something like this:

improvement

Unless there is a strong reason to show the before and after values directly.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16620161

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16620161).

File: packages/components/src/highlight-cards/count-comparison-card.tsx [hunk 1]
```diff
@@ -55,21 +68,47 @@ export function TrendComparison( { count, previousCount }: TrendComparisonProps
...
+		const difference = subtract( count, previousCount ) as number;
+		return (
+			<div className="highlight-card-tooltip-content">
+				<div className="highlight-card-tooltip-counts">
+					{ formatNumber( count, false ) }
+					{ '  ' }
+					{ difference !== 0 && difference !== null && (
+						<span className="highlight-card-tooltip-count-difference">
+							({ difference < 0 ? '-' : '+' }
+							{ formatNumber( Math.abs( difference as number ), false ) })
```
Comment (kangzj) [inline packages/components/src/highlight-cards/count-comparison-card.tsx:93]: It seems formatNumber can handle the sign with option `signdisplay`. Maybe we could leverage that.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#signdisplay

**Review by kangzj — APPROVED** [review]
It works well and looks good 👍 

One minor suggestion no blockers. Thanks for improving this!

**Review by ederrengifo — APPROVED** [review]
This LGTM! Not a blocker for this PR but could we open another issue to be sure all tooltips are consistent? Because I see that likes and comment still have the former format with an icon.

File: packages/components/src/highlight-cards/count-comparison-card.tsx (hunk 1 above)
Comment (jsnmoon) [reply packages/components/src/highlight-cards/count-comparison-card.tsx:93]: Agreed, let's tackle this in a follow-up PR.

**Comment by jsnmoon** [comment]
> Not a blocker for this PR but could we open another issue to be sure all tooltips are consistent?

Excellent callout. I'll get a follow-up PR going for this shortly 👍

**Pull request merged**